app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,215 @@
+
+        $ModelUser      = new ModelUser();
+        $Response       = new Response();
+        $results        = $Response->response(200,'','',true);
+        $name           = Input::get('name');
+        $address        = Input::get('address');
+        $age            = Input::get('age');
+        $avatar         = Input::file('avatar');
+        $created_time   = strtotime( \Carbon\Carbon::now()->toDateTimeString() );
+
+        $data = [
+            'name'          => $name,
+            'address'       => $address,
+            'age'           => $age,
+            'created_time'  => $created_time,
+        ];
+
+        $validator = Validator::make(
+            [
+                'name'      => $name,
+                'address'   => $address,
+                'age'       => $age,
+            ],
+            [
+                'name'      => 'required | max:100',
+                'address'   => 'required | max:300',
+                'age'       => 'required | numeric',
+            ]
+        );
+
+        if ( $validator->fails() ) {
+
+            // The given data did not pass validation
+            $results['meta']['code']    = '0005';
+            $results['meta']['success'] = false;
+            $results['meta']['msg']     = Lang::get('message.web.error.0005');
+
+        } else {
+
+            if ( !empty( $avatar ) ) {
+                
+                $file_name      = $avatar->getClientOriginalName();
+                $file_extension = $avatar->getClientOriginalExtension();
+                $file_real_path = $avatar->getRealPath();
+                $file_mime      = $avatar->getMimeType();
+
+                // image mime types start with "image/"
+                if( substr( $file_mime, 0, 5 ) == 'image' ) {
+
+                    // this is an image
+                    // move file to public path
+                    $full_path_file = 'upload/' . $file_name;
+                    $avatar->move( public_path() . '/upload', $file_name );
+
+                    $data['avatar_path'] = $full_path_file;
+                    
+                } else {
+
+                    // this is not an image
+                    $results['meta']['code']    = '0004';
+                    $results['meta']['success'] = false;
+                    $results['meta']['msg']     = Lang::get('message.web.error.0004');
+                }
+
+            }
+
+            $added_record = $ModelUser->insert( $data );
+
+            if ( $added_record['meta']['success'] ) {
+
+                // insert successfull
+                $results['response']        = $added_record['response'];
+                $results['meta']['msg']     = Lang::get('message.web.success.0001');
+
+            } else {
+
+                $results['meta']['code']    = '0003';
+                $results['meta']['success'] = false;
+                $results['meta']['msg']     = Lang::get('message.web.error.0003');
+            }
+        }
+
+        return json_encode( $results );
```

Comment:
please use this convention: return response()->json($results);

Reply:
applied change